soc/arm/ambiq/apollo4x/soc.c

```diff
@@ -16,7 +17,15 @@
 
 LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
 
+#define AMBIQ_UART_PWRCTRL_ID(uart) AM_HAL_PWRCTRL_PERIPH_UART ## uart
+#define AMBIQ_UART_PWRCTRL_ENABLE(idx) \
+	am_hal_pwrctrl_periph_enable(AMBIQ_UART_PWRCTRL_ID(idx));
+
 void z_arm_platform_init(void)
 {
 	am_bsp_low_power_init();
+
+#define DT_DRV_COMPAT arm_pl011
+	DT_INST_FOREACH_STATUS_OKAY(AMBIQ_UART_PWRCTRL_ENABLE)
+#undef DT_DRV_COMPAT
```

Comment:
-1 on this, please find a better solution (e.g. driver quirk)

Reply:
Moved enabling power for UART to a driver quirk: https://github.com/zephyrproject-rtos/zephyr/blob/4aace128e0d88e660dcdc4a4473f8736927196d0/drivers/serial/uart_pl011_ambiq.h#L54-L68